### PR TITLE
traces: add jaeger exporter

### PR DESCRIPTION
**Pull request by nicoche**
#### PR Description 

WIP

#### Which issue(s) this PR fixes 

https://github.com/grafana/agent/issues/1135

#### PR Checklist

- [x] CHANGELOG updated 
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

File: pkg/traces/config.go [hunk 1]
```diff
@@ -235,7 +248,13 @@ func exporter(rwCfg RemoteWriteConfig) (map[string]interface{}, error) {
 		return nil, errors.New("must have a configured a backend endpoint")
 	}
 
-	headers := map[string]string{}
+	// TODO: investigate
+	// For some reason, jaeger exporter factory makes this field nil when otlp's makes it a 0len map
```
Comment (mapno) [inline pkg/traces/config.go:252]: So, it turns out that the `otlpexporter` sets the headers to an empty map in the [default config](https://github.com/open-telemetry/opentelemetry-collector/blob/v0.41.0/exporter/otlpexporter/factory.go#L49). This is what's causing the diff between otlp and jaeger. Can we document this briefly?

Also, I think it'd be more correct to change the code to
```go
if rwCfg.Format == formatOtel {
```

File: pkg/traces/config.go [hunk 2]
```diff
@@ -147,10 +148,16 @@ const (
 	protocolHTTP    = "http"
 )
 
-// DefaultRemoteWriteConfig holds the default setting	s for a PushConfig.
+const (
+	formatOtel   = "otlp"
```
Comment (mapno) [inline pkg/traces/config.go:152]: Nit: can we change it to `formatOTLP`?

```suggestion
	formatOTLP   = "otlp"
```

**Review by mapno — COMMENTED** [review]
Hey, thanks for the contribution!

I like the direction of the PR. I think `format` is a good choice to configure multiple exporters and addresses the main problem described in https://github.com/grafana/agent/issues/1135.

Left a couple of minor comments.

The next step for this PR would be to document this new parameter in the [config reference](https://github.com/grafana/agent/blob/main/docs/configuration/traces-config.md). Also, please add this change to the changelog.

**Comment by nicoche** [comment]
Hey!
Thanks for taking the time to review this. I'll clean it up.

But before I'd like your opinion on something. I believe that it might be beneficial to deprecate the protocol field.
Exporter formats, as defined by otel, imply a protocol already:
- format [jaegerthrift](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/jaegerthrifthttpexporter) is http
- format [jaeger](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/jaegerexporter) is grpc
- format [otlphttp](https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter/otlphttpexporter) is http
- format [otlp](https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter/otlpexporter) is grpc

I guess it's mostly ok for jaeger and otlp as they both have an an http and a grpc format and we could understand that `protocol:http format:jaeger` refer to jaegerthrift, but regarding formats like [zipkin](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/zipkinexporter), where only http is valid, we could run into this kind of invalid conf:

```yaml
remote_write:
  - endpoint: xxx
    protocol: grpc
    format: zipkin
```

Deprecating `protocol` key in favor of `format` would be clearer imo. Targer confs would be:

```yaml
remote_write:
  - endpoint: xxx
    format: zipkin # or anything from that list actually https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter
```

Thoughts?

**Comment by mapno** [comment]
Hi! Thanks for posting back. These are good points. 

> But before I'd like your opinion on something. I believe that it might be beneficial to deprecate the protocol field.
Exporter formats, as defined by otel, imply a protocol already

I'm not fully convinced in this part. It's my understanding that exporters for different protocols being in separate pkg's is more due to simpler development than being considered each its own format.

Take the OTLP HTTP exporter's definition for instance.

> Exports traces and/or metrics via HTTP using OTLP format.

I think we can run into invalid configs either way. It should be the config code's responsibility to validate them and return meaningful errors.

It's unclear yet how many exporters the Agent will support other than OTLP (gRPC+HTTP) and Jaeger (HTTP), so we will need to list those available in the docs. Similar to listing the different `format` and `protocol` combinations. I don't see a huge gain in that front.

Personally, I'd like to avoid breaking changes for this addition if possible. A new `format` field can be make fully backwards compatible, so I'm reticent to deprecate `protocol`.

**Comment by nicoche** [comment]
Looks fair to me. I've pushed the changes, please let me know if it's missing anything. Once this gets merged, I believe that we could add jaegerthrift / zipkin support pretty easily in subsequent PRs

File: docs/configuration/traces-config.md [hunk 3]
```diff
@@ -59,6 +59,11 @@ remote_write:
     # Only "grpc" is supported in Grafana Cloud.
     [ protocol: <string> | default = "grpc" | supported = "grpc", "http" ]
 
+    # Controls what format to use when exporting traces, in combination with protocol.
+    # protocol/format supported combinations are grpc/otlp, http/otlp and grpc/jaeger
+    # Only grpc/otlp is supported in Grafana Cloud.
```
Comment (nicoche) [inline docs/configuration/traces-config.md:64]: > Only grpc/otlp is supported in Grafana Cloud.

Not sure about that one tbh

Comment (mapno) [reply docs/configuration/traces-config.md:64]: This is correct :)

**Review by mapno — APPROVED** [review]
Sorry for the delay, things are moving slow during these dates.

Thanks for the contribution!

**Comment by nicoche** [comment]
No worries, thanks for taking time to review it. Do you have an idea on when the next release would land?

**Comment by rfratto** [comment]
> No worries, thanks for taking time to review it. Do you have an idea on when the next release would land?

We don't have a standard release schedule (yet) but we were avoiding doing a release during the holiday season. We'll probably have a release within the next few weeks, though right now I don't have a specific date for you.

**Pull request merged**